File: ios/PlayingCard/PlayingCard/data/Card.swift
//
//  Card.swift
//  PlayingCard
//
//  Created by iblis on 28/12/2020.
//

import Foundation

enum CardFigure : Int, CaseIterable, Comparable, Equatable {
  
    
    case TWO = 2, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN
    case JACK = 11, QUEEN, KING
    case ACE = 14
    case JOKER = 15
    
    static func < (lhs: CardFigure, rhs: CardFigure) -> Bool {
        return lhs.rawValue < rhs.rawValue
    }
}

enum CardColor : Int, CaseIterable, Comparable, Equatable{
    case HEARTS = 0, DAIMONDS, CLUBS, SPADES
    
    static func < (lhs: CardColor, rhs: CardColor) -> Bool {
        return lhs.rawValue < rhs.rawValue
    }
}

class Card : Equatable{
    
    var figure:CardFigure
    var color:CardColor
    
    init(figure:CardFigure, color:CardColor){
        self.figure = figure
        self.color = color
    }
    
    
    static func == (lhs: Card, rhs: Card) -> Bool {
        return lhs.figure == rhs.figure && lhs.color == rhs.color
    }
    
}

File: android/PlayingCard/app/src/main/java/com/example/playingcard/data/Card.kt
package com.example.playingcard.data

class Card (val figure: CardFigure, val color: CardColor){

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Card

        if (figure != other.figure) return false
        if (color != other.color) return false

        return true
    }

    override fun hashCode(): Int {
        var result = figure.hashCode()
        result = 31 * result + color.hashCode()
        return result
    }
}